lib/roo/excelx/cell/datetime.rb

```diff
@@ -92,18 +96,9 @@ def parse_date_or_time_format(part)
           '0' => '%1N'    # Fractional Seconds: tenths.
         }
 
-        def create_datetime(base_date, value)
-          date = base_date + value.to_f.round(6)
-          datetime_string = date.strftime('%Y-%m-%d %H:%M:%S.%N')
-          t = round_datetime(datetime_string)
-
-          ::DateTime.civil(t.year, t.month, t.day, t.hour, t.min, t.sec)
-        end
-
-        def round_datetime(datetime_string)
-          /(?<yyyy>\d+)-(?<mm>\d+)-(?<dd>\d+) (?<hh>\d+):(?<mi>\d+):(?<ss>\d+.\d+)/ =~ datetime_string
-
-          ::Time.new(yyyy, mm, dd, hh, mi, ss.to_r).round(0)
+        def create_datetime(base_timestamp, value)
+          timestamp = (base_timestamp + (value.to_f.round(6) * SECONDS_IN_DAY)).round(0)
```

Comment:
I've compared the o/p of new and old implementation.

I/P value edge case (if your timezone is IST) :
```
2193.0..2193.00613  # 8min 20sec long
15250.0..15250.04166 # 1 hour long
15585.0..15585.04166 # 1 hour long
```
The above range of values are overlap of timezone's (IST) offset changes.

At the moment I'm not sure if the new implementation is right or older one. I'll have to test a bit more to verify that. If it is the older implementation which wrong we can go ahead with this PR otherwise need to find some workaround